src/OpenTelemetry.AutoInstrumentation/Loading/AspNetCoreInitializer.cs

```diff
@@ -16,22 +16,29 @@
 
 #if NETCOREAPP3_1_OR_GREATER
 using System;
+using OpenTelemetry.AutoInstrumentation.Plugins;
 
 namespace OpenTelemetry.AutoInstrumentation.Loading;
 
 internal class AspNetCoreInitializer : InstrumentationInitializer
 {
-    public AspNetCoreInitializer()
+    private readonly PluginManager _pluginManager;
+
+    public AspNetCoreInitializer(PluginManager pluginManager)
         : base("Microsoft.AspNetCore.Http")
     {
+        _pluginManager = pluginManager;
     }
 
     public override void Initialize(ILifespanManager lifespanManager)
     {
         var instrumentationType = Type.GetType("OpenTelemetry.Instrumentation.AspNetCore.AspNetCoreInstrumentation, OpenTelemetry.Instrumentation.AspNetCore");
         var httpInListenerType = Type.GetType("OpenTelemetry.Instrumentation.AspNetCore.Implementation.HttpInListener, OpenTelemetry.Instrumentation.AspNetCore");
 
-        var httpInListener = Activator.CreateInstance(httpInListenerType, args: new OpenTelemetry.Instrumentation.AspNetCore.AspNetCoreInstrumentationOptions());
+        var options = new OpenTelemetry.Instrumentation.AspNetCore.AspNetCoreInstrumentationOptions();
+        _pluginManager.ConfigureOptions(options);
```

Comment:
This could crash the application at runtime if the plugin throws an exception. Right?

Reply:
Seems something is handling the exception. Somehow the ASP.NET Core app reports exception (in debug mode), continues and produces spans 🤔